src/main/java/net/minecraftforge/client/model/ForgeFaceData.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client.model;
+
+import java.util.function.Function;
+
+import com.mojang.datafixers.util.Either;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+import net.minecraft.client.renderer.block.model.BlockElement;
+import net.minecraft.client.renderer.block.model.BlockElementFace;
+import net.minecraft.util.ExtraCodecs;
+
+/**
+ * Holds extra data that may be injected into a face.<p>
+ * Used by {@link ItemLayerModel}, {@link BlockElement} and {@link BlockElementFace}
+ * 
+ * @param color Color in ARGB format
+ * @param blockLight Block Light for this face from 0-15 (inclusive)
+ * @param skyLight Sky Light for this face from 0-15 (inclusive)
+ */
+public record ForgeFaceData(int color, int blockLight, int skyLight)
+{
+
+    public static final ForgeFaceData DEFAULT = new ForgeFaceData(0xFFFFFFFF, 0, 0);
+
+    public static final Codec<Integer> COLOR = new ExtraCodecs.EitherCodec<>(Codec.INT, Codec.STRING).xmap(
+            either -> either.map(Function.identity(), str -> (int) Long.parseLong(str, 16)),
```

Comment:
Reason is likely because `0xF#######` is not a valid int, you would instead be forced to write `-0x7#######` which is a lot less readable. Long allows the 8 digit values.

Reply:
That is indeed the reason - if you try to parse `FFFFFFFF` with parseInt you get a NumberFormatException.